### PR TITLE
Fix for 4-d variable output in standard memory mode

### PR DESCRIPTION
**Pull request by anewman89**
Issuing pull request for potential bug fix in 4-d variable output for `standard` memory mode.

### CODE REVIEW

File: tonic/models/vic/vic2netcdf.py [hunk 1]
```diff
@@ -424,11 +417,9 @@ def nc_fields(self, fields, y_x_dims, precision):
                                                           zlib=False)
 
                 if 'units' in field:
-                    self.fields[name].long_name = name.encode()
-                    self.fields[name].coordinates = 'lon lat'.encode()
+                    self.fields[name].long_name = name
+                    self.fields[name].coordinates = 'lon lat'
                     for key, val in field.items():
-                        if isinstance(val, str):
-                            val = val.encode()
```
Comment (jhamman) [inline tonic/models/vic/vic2netcdf.py]: You'll need to undo all the changes to the string encoding that came in here.  You can do that manually or by using something like source tree.

The reason this happened is that you copied in an old version of this file (that wasn't in git) into the current develop branch so you've just undone all the changes we've recently made to the file, that git has recorded.  Does that make sense?

**Comment by jhamman** [comment]
fixes #24

File: tonic/models/vic/vic2netcdf.py [hunk 2]
```diff
@@ -456,16 +447,31 @@ def nc_add_data_to_array(self, point):
...
-                varshape = self.f.variables[name].shape[1]
-                for i in pyrange(varshape):
-                    sn = name + str(i)
-                    self.f.variables[name][:, i, ys,
-                                           xs] = p.df[sn].values[self.slice]
+        for name in self.three_dim_vars:
+            yc = 0
+            dd = np.zeros(self.f.variables[name][:,ys,xs].shape)
+            for p in points:
+                dd[:,yc] = p.df[name].values[self.slice]
```
Comment (jhamman) [inline tonic/models/vic/vic2netcdf.py]: This seems like we're just writing the same data to all of these points.

File: tonic/models/vic/vic2netcdf.py [hunk 3]
```diff
@@ -456,16 +447,31 @@ def nc_add_data_to_array(self, point):
...
+                yc = yc + 1
+
+            self.f.variables[name][:,ys,xs] = dd
+            del dd
```
Comment (jhamman) [inline tonic/models/vic/vic2netcdf.py]: The python garbage collector will handle this for you.  It is almost never necessary to use the `del` statement in python.

File: tonic/models/vic/vic2netcdf.py [hunk 4]
```diff
@@ -456,16 +447,31 @@ def nc_add_data_to_array(self, point):
...
-                for i in pyrange(varshape):
-                    sn = name + str(i)
-                    self.f.variables[name][:, i, ys,
-                                           xs] = p.df[sn].values[self.slice]
+        for name in self.three_dim_vars:
+            yc = 0
+            dd = np.zeros(self.f.variables[name][:,ys,xs].shape)
+            for p in points:
+                dd[:,yc] = p.df[name].values[self.slice]
+                yc = yc + 1
```
Comment (jhamman) [inline tonic/models/vic/vic2netcdf.py]: Python allow you to enumerate your for loop.  You can get rid of all the `yc` logic with this for loop syntax: `for yc, p in enumerate(points):`.

**Comment by jhamman** [comment]
I'll have to look at this a bit more.  I don't love either implementation so I'll need to think on it.  I gave you some comments to get you started cleaning it up a bit though.  Thanks.

**Comment by bartnijssen** [comment]
@anewman89 : Just ran across this - is this still an active PR?

**Comment by anewman89** [comment]
I never got anywhere further with this.  In fact, my fix is likely buggy as I got some large disk I/O issues when I ran it on Yellowstone at NCAR.  So lets close this specific request without a merge, I can make the original reason for the pull request an issue.

**Comment by bartnijssen** [comment]
Agreed - open an issue and then close this PR

**Pull request closed without merging**